driver_config.go

```
@@ -21,19 +24,264 @@ const driverConfigVersion = 1
+// configuration. 32KiB is generous for a configuration report and matches the
+// limit csharp-driver and java-driver use for the same report.
 const maxDriverConfigSize = 32 * 1024
 
-// driverConfigReport is the value sent under driverConfigStartupKey.
-// For now only the schema version is reported; configuration groups will be
-// added in a follow-up change.
+// driverConfigReport is the value sent under driverConfigStartupKey, conforming
+// to docs/driver-config-schema.json. It groups the effective configuration the
+// way the schema does: connection-level settings, control-plane (system query
+// and schema agreement) settings, and query-execution defaults.
 type driverConfigReport struct {
```

golangci-lint (2.12.2)

[error] fieldalignment: struct with 216 pointer bytes could be 200

(govet)